sdk/src/types/block/output/basic.rs

```diff
@@ -151,8 +160,73 @@ impl BasicOutputBuilder {
+    /// the additional unlock condition that will return the remainder to the provided `return_address`.
+    pub fn with_sufficient_storage_deposit(
+        mut self,
+        return_address: impl Into<Address>,
+        rent_structure: RentStructure,
+        token_supply: u64,
+    ) -> Result<Self, Error> {
+        Ok(match self.amount {
+            OutputBuilderAmount::Amount(amount) => {
+                let return_address = return_address.into();
+                // Get the current rent requirement
+                let rent_cost = self.rent_cost(rent_structure);
+                // Check whether we already have enough funds to cover it
+                if amount < rent_cost {
+                    // Get the projected rent cost of the return output
+                    let return_rent_cost = Self::new_with_amount(0)
+                        .add_unlock_condition(AddressUnlockCondition::new(return_address))
+                        .rent_cost(rent_structure);
+                    // Add a temporary storage deposit unlock condition so the new rent requirement can be calculated
+                    self = self.add_unlock_condition(StorageDepositReturnUnlockCondition::new(
+                        return_address,
+                        0,
+                        token_supply,
+                    )?);
+                    // Get the rent cost of the output with the added storage deposit return unlock condition
+                    let rent_cost_with_sdruc = self.rent_cost(rent_structure);
+                    // If the return rent cost and amount are less than the required min
+                    let (amount, sdruc_amount) = if rent_cost_with_sdruc >= return_rent_cost + amount {
+                        // Then sending rent_cost_with_sdruc covers both minimum requirements
+                        (rent_cost_with_sdruc, rent_cost_with_sdruc - amount)
+                    } else {
+                        // Otherwise we must use the total of the return minimum and the original amount
+                        // which is unfortunately more than the rent_cost_with_sdruc
+                        (return_rent_cost + amount, return_rent_cost)
+                    };
+                    // Add the required storage deposit unlock condition and the additional rent amount
+                    self.with_amount(amount)
+                        .replace_unlock_condition(StorageDepositReturnUnlockCondition::new(
+                            return_address,
+                            sdruc_amount,
+                            token_supply,
+                        )?)
+                } else {
+                    self
+                }
+            }
+            OutputBuilderAmount::MinimumStorageDeposit(_) => self,
+        })
+    }
+
+    pub fn min_storage_deposit_amount(&self, rent_structure: RentStructure, token_supply: u64) -> Result<u64, Error> {
```

Comment:
From the name it's very unclear to me why it calls `with_sufficient_storage_deposit()` and not just returns the minimum required storage deposit amount without potentially adding a SDRUC

Reply:
It clones the value, so it does not add a SDRUC. The reason I call the other fn is because most of the functionality would have be re-implemented either way.

Reply:
It doesn't add the SDRUC to the initial builder, but it adds it before calculating the amount if not enough amount is there
This function doesn't return the min_storage_deposit_amount as the name suggests, but just the amount the output has if higher than the rent_cost or the rent cost it would need with a SDRUC

Reply:
Exactly, which is the min storage deposit amount?

Reply:
If min rent cost for an output is 42600, but you call this on an output with 100000000 as amount, it would return 100000000

Reply:
Yes, that is correct. Because that is the minimum amount needed.

Reply:
I don't get how you can call this a `minimum`. To me its just the amount of the output with potentially a SDRUC if needed (only in this case `minimum` makes sense to me)

Reply:
Think of it like this
```rust
current_amount.min(needed_amount)
```
It's not exactly that of course but that's what I mean by minimum